### PR TITLE
fix host ip retrieving for package contrib/trace/jaeger when there is no intranet ip in host

### PR DESCRIPTION
**Pull request by gqcn**

### CODE REVIEW

File: contrib/trace/jaeger/jaeger.go [hunk 1]
```diff
@@ -48,10 +49,23 @@ func Init(serviceName, endpoint string) (*trace.TracerProvider, error) {
...
 		return nil, err
 	}
+	if len(intranetIpArray) == 0 {
+		var ipArray []string
+		if ipArray, err = gipv4.GetIpArray(); err != nil {
+			return nil, err
+		}
+		if len(ipArray) > 0 {
+			hostIp = ipArray[0]
+		}
```
Comment (houseme) [inline contrib/trace/jaeger/jaeger.go:64]: 如果还是没有 IP 呢？

File: contrib/trace/jaeger/jaeger.go [hunk 2]
```diff
@@ -48,10 +56,27 @@ func Init(serviceName, endpoint string) (*trace.TracerProvider, error) {
 			jaeger.WithAgentHost(host), jaeger.WithAgentPort(port),
 		)
 	}
-	hostIP, err := gipv4.GetIntranetIp()
+	// Try retrieving host ip for tracing info.
+	var hostIp string
+	intranetIpArray, err := gipv4.GetIntranetIpArray()
```
[the remark this answers is not in the document]
Comment (gqcn) [reply contrib/trace/jaeger/jaeger.go]: done

**Comment by houseme** [comment]
LGTM

**Review by houseme — APPROVED** [review]
LGTM

**Pull request merged**